translation.rst

```diff
@@ -292,6 +292,35 @@ To manage these situations, Symfony follows the `ICU MessageFormat`_ syntax by
 using PHP's :phpclass:`MessageFormatter` class. Read more about this in
 :doc:`/translation/message_format`.
 
+Translatable Objects
+--------------------
+
+Sometimes you may want to create a message, but at the time of creation aren't
+sure how it would be translated. For example, it could be translated multiple
+times if intended to be displayed to multiple users.
```

Comment:
In my POV the main benefit of to translate things this way is that you can "create" (prepare) translations without needing a dependency on the router itself.

This way for example entities can return translated labels, etc..

Maybe we want to mention that?

Reply:
Added some wording around it. Not entirely happy. Open to suggestions. =)